packages/mco/components/modals/app-manage-policies/helper/policy-config-viewer.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import { pluralize } from '@odf/core/components/utils';
+import { REPLICATION_DISPLAY_TEXT } from '@odf/mco/constants';
+import { parseSyncInterval } from '@odf/mco/utils';
+import { fromNow } from '@odf/shared/details-page/datetime';
+import { SingleSelectDropdown } from '@odf/shared/dropdown/singleselectdropdown';
+import { Labels } from '@odf/shared/labels/labels';
+import { getName } from '@odf/shared/selectors';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+} from '@odf/shared/status/icons';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { StatusIconAndText } from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import {
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  SelectOption,
+  Text,
+} from '@patternfly/react-core';
+import { DRPlacementControlType, DataPolicyType } from '../utils/types';
+import '../style.scss';
+
+const getDropdownOptions = (
+  placements: string[],
+  defaultOptionText: string
+): JSX.Element[] => [
+  <SelectOption key={defaultOptionText} value={defaultOptionText} />,
+  ...placements?.map((placement) => (
+    <SelectOption key={placement} value={placement} />
+  )),
+];
+
+const getLabels = (
+  placementControlInfo: PlacementControlMap,
+  selected: string,
+  isDefaultSelected: boolean
+): string[] =>
+  isDefaultSelected
+    ? // Aggregate all labels
+      Object.values(placementControlInfo)?.reduce(
```

Comment:
we are overusing `?.` everywhere (in general in mco)... we can keep default value of `placementControlInfo` as `{}` and avoid using `?.` here...
same is true for other places as well, where it is possible to handle the cases like that...
```suggestion
      Object.values(placementControlInfo).reduce(
```

Reply:
i will try to re-evaluate on this PR, i will check and remove it any unnecessary options symbol used